MCP server fo CHR (#725)
JetBrains/kotlin-multiplatform-dev-docs: topics/tools/compose-hot-reload.md

Congratulations! You've seen Compose Hot Reload in action. Now you can experiment with changing text, images, formatting, 
UI structure, and more, without having to restart the desktop run configuration after every change.

## MCP server for AI agents
<primary-label ref="Experimental"/>

> Starting with version 1.2.0-alpha01, Compose Hot Reload includes a built-in

Reviewer: Since CHR is bundled with CMP, I'm not sure how do I experiment this experiment. Do I need to explicitly specify this 1.2.0-alpha01 version somewhere, or does CMP 1.12.0 include it from the very start? How

[Model Context Protocol (MCP)](https://modelcontextprotocol.io/) server. 

> The MCP server lets AI coding agents interact with your running Compose application: 

Reviewer: ```suggestion
The MCP server lets AI coding agents interact with an application that uses Compose Hot Reload:
```

> trigger Compose Hot Reload, see the rendered UI, inspect the UI structure, simulate user input, and read runtime logs.

Reviewer: ```suggestion
trigger UI updates, see the rendered UI, inspect the semantic structure, simulate user input, and read runtime logs.
```

For applications with multiple windows, the agent can list windows and target any of them.

This closes the feedback loop for AI agents when editing Compose code. 

> Instead of relying on you to manually check the result after every edit, 

Reviewer: ```suggestion
Instead of relying on you to visually check the result after every edit, 
```

?

Author: It's not just about visuals -- the agent can work with logs, for example

the agent can iterate on your code autonomously and verify each change.

### Connect an AI agent

To let an AI agent interact with your Compose application:

1. Start the MCP server alongside your application using the Gradle task:
    ```shell
    ./gradlew :composeApp:hotMcpServerJvm
    ```

>     For a custom JVM target named `desktop`, the task would be `:composeApp:hotMcpServerDesktop`.

Reviewer: Does this mean that `hotMcpServerJvm` is an example? Or is it that `hotMcpServerJvm` is always there, but for additional targets Gradle imports additional tasks?

2. Point your AI agent's MCP client configuration to the Gradle task. For example, in `.mcp.json`:
    ```json
    {
      "mcpServers": {
        "compose-hot-reload": {
          "command": "./gradlew",
          "args": [
            "--no-daemon",
            "--quiet",
            "--console=plain",

>             "hotMcpServer"

Reviewer: This is neither `hotMcpServerJvm` nor `hotMcpServerDesktop`, so I'm lost now :)

          ]
        }
      }
    }
    ```

### Available MCP tools

The MCP server exposes a range of tools the agent can invoke, including:

* `reload` — recompiles the project and hot-reloads the changed classes.
* `take_screenshot` — captures the current state of an application window.

> * `get_semantic_tree` — returns the Compose semantic/accessibility tree, so the agent can understand the UI structure.

Reviewer: ```suggestion
* `get_semantic_tree` — returns the Compose semantics tree, so the agent can understand the UI structure.
```

Maybe we can link the accessibility page here, but I think it's called "semantics tree" in Compose and "accessibility tree" in iOS.
